src/python/gudhi/random_point_generators.cc

```diff
@@ -0,0 +1,68 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hind Montassif
+ *
+ *    Copyright (C) 2021 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#include <gudhi/random_point_generators.h>
+
+#include <CGAL/Epick_d.h>
+
+namespace py = pybind11;
+
+
+typedef CGAL::Epick_d< CGAL::Dynamic_dimension_tag > Kern;
+
+template <typename Kernel>
+py::array_t<double> generate_points_on_sphere(py::object num_points, py::object dim, py::object radius) {
+    int npoints = num_points.cast<int>();
+    int d = dim.cast<int>();
+    double rad  = radius.cast<double>();
+    
+    py::gil_scoped_release release;
+
+    auto points_generated = Gudhi::generate_points_on_sphere_d<Kernel>(npoints, d, rad);
+    
+    py::gil_scoped_acquire acquire;
+    
+    py::array_t<double> points({npoints, d});
+ 
+    py::buffer_info buf = points.request();
+
+    double *ptr = static_cast<double *>(buf.ptr);
+
+    assert(npoints == buf.shape[0]);
+    assert(d == buf.shape[1]);
```

Comment:
I am not well placed to comment on that because I used assert in `_tomato.cc`, but I think we mostly use GUDHI_CHECK in C++ code. Vincent?

Reply:
I was actually wondering why changing the expression in assert, deliberately to false, did nothing.
The answer was: 
https://github.com/GUDHI/gudhi-devel/blob/f21240a4c4e0fad25193dc0d409deee9715e0aae/src/common/include/gudhi/Debug_utils.h#L20-L28

So both assert and GUDHI_CHECK do nothing in release mode.

@VincentRouvreau ?

Reply:
Yes, I think it is better to use GUDHI_CHECK and yes, it throws an exception only in debug mode:
https://github.com/GUDHI/gudhi-devel/blob/f21240a4c4e0fad25193dc0d409deee9715e0aae/src/common/include/gudhi/Debug_utils.h#L15-L18
For exceptions, please use the ones from the standard (I shall write this in the developer guide).
If you want to test, you can modify your setup.py (in build/src/python) by adding:
```python
extra_compile_args=['-DNDEBUG', ...]
```
Maybe we could add this compile args when compiling in debug mode (from CMake).

Reply:
> extra_compile_args=['-DNDEBUG', ...]

NDEBUG means **not** debug, so you want to undefine it instead if you mean to test debug mode.

Reply:
Oops, sorry ! It was `-U` and not `-D`
```
extra_compile_args=['-UNDEBUG', ...]
```
It seems to work for me.
We could add this feature to debug mode.

Reply:
I think the basic flags come from python, so maybe if I install `python3-dbg` and pass `-DPython_ADDITIONAL_VERSIONS=3d` I get the plugin in debug mode as well? I haven't tried, it may require installing a debug version of all the dependencies... Your version could be useful to get assertions with a plugin compiled for release-mode python. We can probably assume that the extra_ arguments are passed after the regular ones.